packages/medusa/src/services/batch-job.ts

```diff
@@ -0,0 +1,115 @@
+import { EntityManager } from "typeorm"
+import { BaseService } from "medusa-interfaces"
+import { MedusaError } from "medusa-core-utils"
+
+import { BatchJob } from "../models"
+import { BatchJobRepository } from "../repositories/batch-job"
+import { BatchJobStatus, FilterableBatchJobProps } from "../types/batch-job"
+import { FindConfig } from "../types/common"
+import EventBusService from "./event-bus"
+
+type InjectedContainer = {
+  manager: EntityManager
+  eventBusService: EventBusService
+  batchJobRepository: typeof BatchJobRepository
+}
+
+class BatchJobService extends BaseService<BatchJobService> {
```

Comment:
you can extend the `TransactionBaseService` now :)

Reply:
the actual BaseService is not template

Reply:
done 😉